dbms/src/Core/Field.h

```diff
@@ -122,7 +122,20 @@ class DecimalField
         }
         if (cnt == 0)
             cnt = 1;
-        return cnt;
+        return std::max(cnt, scale);
+    }
+
+    /// In TiFlash there are 4 subtype of decimal:
+    /// Decimal32, Decimal64, Decimal128 and Decimal256
+    /// they are not compatible with each other. So a DecimalField<Decimal32>
+    /// can not be inserted into a decimal column with DecimalType<Decimal64>
+    /// getPrecWithCurrentDecimalType will return the prec that fit
+    /// current decimal type, that is to say, current DecimalField can be
+    /// inserted into a decimal column with type `Decimal(getPrecWithCurrentDecimalType, getScale)`
+    UInt32 getPrecWithCurrentDecimalType() const
+    {
+        auto raw_prec = getPrec();
```

Comment:
Will `getPrec() > maxDecimalPrecision<T>`?

Reply:
No, basically, `getPrec()` is getting the precision of a literal of type `T`, so `getPrec() > maxDecimalPrecision<T>` should not happen.

Reply:
add `assert(getPrec() <= maxDecimalPrecision<T>)`?